.github/workflows/build-and-push-minimal.yml

```diff
@@ -0,0 +1,84 @@
+name: Build and push s2i-python minimal images to Quay.io registry
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build-and-push-minimal:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        version: [3.9-minimal]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: true
+
+      - name: Get base image name
+        id: base-image-name
+        run: |
+          # This command returns row with BASE_IMAGE_NAME
+          row=$(grep "BASE_IMAGE_NAME" Makefile)
+          # Return only base image name
+          BASE_IMAGE_ROW=${row/BASE_IMAGE_NAME = /}
+          echo ::set-output name=image_name::$BASE_IMAGE_ROW
+
+      - name: Install python3 packages needed by distgen
+        run: |
+          sudo apt-get update -y && sudo apt-get install -y python3 python3-pip python3-jinja2 python3-yaml python3-distro
```

Comment:
I believe that you need only python3 and python3-pip. Jinja, pyyaml and distro (+six) are dependencies of distgen and they are all installed together with distgen from PyPI.